src/cmd/verify.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright 2018-2022 Parity Technologies (UK) Ltd.
+// This file is part of cargo-contract.
+//
+// cargo-contract is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// cargo-contract is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with cargo-contract.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::{
+    cmd::{
+        build::{
+            execute,
+            ExecuteArgs,
+        },
+        metadata::BuildInfo,
+    },
+    workspace::ManifestPath,
+    BuildArtifacts,
+};
+
+use anyhow::Result;
+use contract_metadata::{
+    ContractMetadata,
+    SourceWasm,
+};
+
+use std::{
+    fs::File,
+    io::prelude::Read,
+    path::PathBuf,
+};
+
+/// Checks if a contract in the given workspace matches that of a reference contract.
+#[derive(Debug, clap::Args)]
+#[clap(name = "verify")]
+pub struct VerifyCommand {
+    /// Path to the `Cargo.toml` of the contract to verify.
+    #[clap(long, parse(from_os_str))]
+    manifest_path: Option<PathBuf>,
+    /// The reference Wasm contract (`*.contract`) that the workspace will be checked against.
+    contract: PathBuf,
+}
+
+impl VerifyCommand {
+    pub fn run(&self) -> Result<()> {
+        let manifest_path = ManifestPath::try_from(self.manifest_path.as_ref())?;
+
+        // 1. Read the given metadata, and pull out the `BuildInfo`
+        let mut file = File::open(&self.contract)?;
+        let mut contents = String::new();
+        file.read_to_string(&mut contents)?;
+
+        let metadata: ContractMetadata = serde_json::from_str(&contents)?;
```

Comment:
Can also do `from_reader` to avoid the intermediate `String` e.g. https://github.com/paritytech/cargo-contract/blob/master/transcode/src/lib.rs#L171